src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupCodeHelpers.cs

```diff
@@ -174,8 +174,12 @@ private static unsafe void InitializeStatics(IntPtr gcStaticRegionStart, int len
             IntPtr gcStaticRegionEnd = (IntPtr)((byte*)gcStaticRegionStart + length);
             for (IntPtr* block = (IntPtr*)gcStaticRegionStart; block < (IntPtr*)gcStaticRegionEnd; block++)
             {
-                object obj = RuntimeImports.RhNewObject(new EETypePtr(*block));
-                *block = RuntimeImports.RhHandleAlloc(obj, GCHandleType.Normal);
+                IntPtr* pBlock = (IntPtr*)*block;
+                if (((*pBlock).ToInt64() & 0x1L) == 1)
```

Comment:
Might be good to add the comment extracted from the commit log somewhere in the code. I'm suggesting:
```
// Prevent statics from being initialized multiple times by only initializing GCStatics whose lowest bit is set. It is
// masked off before passing to RhAllocate.  After init is done, the GCStaticDesc's pointer is overwritten with the
// properly set up GC Static info.  This mechanism allows GCStatics referred to by multiple contributing modules
// (ie, a common generic instantiation) to only have its statics initialized once.
```